### PR TITLE
[USH-3280] Retain order of DD group properties

### PR DESCRIPTION
**Pull request by akashkj**
## Product Description
Retain ordering of data dictionary groups and its properties in case data UI. Exception of `Unrecognized` and `Uncategorized` groups, that are to be at the bottom, same as earlier.

## Technical Summary
https://dimagi-dev.atlassian.net/browse/USH-3280

Remove `sorted()` while creating `DisplayConfig` objects for tabular representation of properties and while grouping properties by group name.

## Feature Flag
`data_dictionary` and `dd_case_data`

### Safety story
Local testing
image

### Automated test coverage
Added unit tests

### QA Plan
NA

### Migrations
NA

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/reports/standard/cases/case_data.py [hunk 1]
```diff
@@ -298,7 +297,7 @@ def _get_dd_props_by_group(domain, case_type):
         ret[prop.group_name].append(prop)
 
     uncategorized = ret.pop('', None)
-    for group, props in sorted(ret.items()):
+    for group, props in ret.items():
```
Comment (esoergel) [inline corehq/apps/reports/standard/cases/case_data.py:300]: Does this return the items in the order they were created or the `index`?  I'd expect to see a `.order_by('index')` in there.  That is, if you rearrange the properties in the DD, does this use the new ordering?  (Likewise for group ordering)

File: corehq/apps/reports/tests/test_case_data.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        """
+        Sample result
+        [{'name': 'group1', 'rows': [[{'expr': 'prop1', 'name': 'prop1', 'description': '', 'value': '---',
+        'has_history': True}, {'expr': 'prop2', 'name': 'prop2', 'description': '', 'value': '---',
+        'has_history': True}]]}, {'name': 'group2', 'rows': [[{'expr': 'prop3', 'name': 'prop3',
+        'description': '', 'value': '---', 'has_history': True}, {'expr': 'prop4', 'name': 'prop4',
+        'description': '', 'value': '---', 'has_history': True}]]}]
+        """
+        idx = 0
+        for group_data in result:
```
Comment (esoergel) [inline corehq/apps/reports/tests/test_case_data.py:75]: nit: you can zip these two things together so you don't need to worry about indices
```suggestion
        self.assertEqual(len(result), len(expected_result))
        for group_actual, group_expected in zip(result, expected_result):
```

File: corehq/apps/reports/standard/cases/case_data.py (hunk 1 above)
Comment (akashkj) [reply corehq/apps/reports/standard/cases/case_data.py:300]: https://github.com/dimagi/commcare-hq/pull/33098/commits/86978863149b895ef032785c6a59520c344c614c

Have added the sorting on group index and property index(Similar to [this](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/data_dictionary/views.py#L63)). Somehow previous change was also giving correct ordering for test data.

Comment (esoergel) [reply corehq/apps/reports/standard/cases/case_data.py:300]: Thanks - I was wondering about that too - a theory is that perhaps the default order is by date saved, and when editing the DD via the UI, we save all properties in sequence, meaning that the save order is the same as the index order.  Definitely safer and more explicit to add an `.order_by`, though.

**Review by esoergel — APPROVED** [review]

File: corehq/apps/reports/standard/cases/case_data.py [hunk 3]
```diff
@@ -293,15 +293,15 @@ def _get_dd_props_by_group(domain, case_type):
             case_type__domain=domain,
             case_type__name=case_type,
             deprecated=False,
-    ):
+    ).select_related('group_obj').order_by('group_obj__index', 'index'):
```
[the remark this answers is not in the document]
Comment (mkangia) [reply corehq/apps/reports/standard/cases/case_data.py:296]: @pxwxnvermx I vaguely remember us talking about renaming group_obj to group in the PR where you originally added groups? That is yet to happen yes?

[the remark this answers is not in the document]
Comment (pxwxnvermx) [reply corehq/apps/reports/standard/cases/case_data.py:296]: @mkangia Yes, that change is yet to happen.

**Comment by mkangia** [comment]
:100: 

Thanks for jumping in for the review on this one @esoergel

**Pull request merged**